plugins/ui-update/components/AvailableUpdateRow.tsx

```diff
@@ -0,0 +1,103 @@
+import { Trans } from "@lingui/macro";
+import * as React from "react";
+
+import ConfigurationMapLabel from "#SRC/js/components/ConfigurationMapLabel";
+import ConfigurationMapRow from "#SRC/js/components/ConfigurationMapRow";
+import ConfigurationMapValue from "#SRC/js/components/ConfigurationMapValue";
+import {
+  UIAction,
+  UIActions,
+  UIActionType
+} from "#PLUGINS/ui-update/types/UIAction";
+import { FormattedPackageVersion } from "#PLUGINS/ui-update/types/FormattedPackageVersion";
+
+export interface AvailableUpdateRowProps {
+  latestVersion: FormattedPackageVersion | null;
+  uiAction: UIAction;
+  onUpdateClick: (version: string) => void;
+}
+
+class AvailableUpdateRow extends React.PureComponent<
+  AvailableUpdateRowProps,
+  {}
+> {
+  private static renderActionContent(uiAction: UIAction): React.ReactNode {
+    if (uiAction.type !== UIActionType.Update) {
+      return <Trans>Start Update</Trans>;
+    }
+    switch (uiAction.action) {
+      case UIActions.Completed:
+        return <Trans>Update Complete</Trans>;
+      case UIActions.Started:
+        return <Trans>Updating...</Trans>;
+      case UIActions.Error:
+        return <Trans>Update Failed!</Trans>;
+      default:
+        return <Trans>Start Update</Trans>;
+    }
+  }
+
+  private static actionDisabled(uiAction: UIAction): boolean {
```

Comment:
`static` could be a sign that the method doesn't belong to the class.

Reply:
I can extract the update & rollback buttons to their own components. 

I prefer to have this logic in the class as you can see how the content/disabled states are determined. Which is why I left them here even as static functions.